### PR TITLE
 #1955 Fix foreign key generation in MetadataExp

### PR DESCRIPTION
**Pull request by balazs-zsoldos**
I collect the foreign keys that should be generated
into a set and only add them to the classModel if
there is any foreign key.

### CODE REVIEW

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java [hunk 1]
```diff
@@ -356,11 +356,16 @@ private void handleTable(DatabaseMetaData md, ResultSet tables) throws SQLExcept
...
                 if (!foreignKeyData.isEmpty()) {
+                    Collection<ForeignKeyData> foreignKeysToGenerate = new HashSet<ForeignKeyData>();
                     for (ForeignKeyData fkd : foreignKeyData.values()) {
                         if (namingStrategy.shouldGenerateForeignKey(schemaAndTable, fkd)) {
-                            classModel.getData().put(ForeignKeyData.class, foreignKeyData.values());
+                            foreignKeysToGenerate.add(fkd);
                         }
                     }
+
+                    if (foreignKeysToGenerate.size() > 0) {
```
Comment (Shredder121) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java]: Could you change this to:
```java
if (!foreignKeysToGenerate.isEmpty()) 
```

**Comment by Shredder121** [comment]
Yeah, this is indeed an issue.
Thanks for opening up a PR.
I left a comment in the diff, but I don't know whether others have comments or not.

File: querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java (hunk 1 above)
Comment (balazs-zsoldos) [inline querydsl-sql-codegen/src/main/java/com/querydsl/sql/codegen/MetaDataExporter.java]: @Shredder121 Changed it.

**Comment by timowest** [comment]
Maybe still squash the commits

**Comment by balazs-zsoldos** [comment]
If I look at the _Files changed_ tab, I see that the isEmpty function is used now.

**Comment by timowest** [comment]
I meant to combine the two commits into one. The end result is fine.

**Comment by balazs-zsoldos** [comment]
@timowest  I see. I have never done it before, but I think I managed to merge the two commits.

**Comment by timowest** [comment]
Looks good, thanks a lot for the contribution @balazs-zsoldos!

**Comment by Shredder121** [comment]
Look good to me as well.
I'll merge it then.
@balazs-zsoldos Thanks for the contribution!

**Pull request merged**